week-1/implement/max.js

```diff
@@ -0,0 +1,34 @@
+function findMaxNumber(numbers) {
+    if (isEmptyArray(numbers)) {
+      return -Infinity;
+    }
+  
+    let maximumNumber = findInitialMax();
+  
+    for (const element of numbers) {
+      if (isNumeric(element)) {
+        maximumNumber = compareAndUpdateMax(maximumNumber, element);
+      }
+    }
+  
+    return maximumNumber;
+  }
+  
+  function isEmptyArray(array) {
+    return array.length === 0;
+  }
+  
+  function findInitialMax() {
+    return -Infinity;
+  }
+  
+  function isNumeric(value) {
+    return typeof value === 'number' && !isNaN(value);
+  }
+  
+  function compareAndUpdateMax(currentMax, newNumber) {
+    return Math.max(currentMax, newNumber);
+  }
+  
```

Comment:
Again good work, but the amount of functions you wrote for a simple task is more than you need also it makes a bit difficult for a reviewer to understand. can you try to reduce these functions?
 Don't get me wrong the approach you are doing is great. But not for very simple serio like this